API/controller/mealController.js

```diff
@@ -0,0 +1,114 @@
+import mealData from "../utils/mealData";
+import mealModel from "../model/mealModels";
+
+const MealController = {
+  fetchAllMeals(req, res) {
+    return res.status(200).json({
+      status: 200,
+      data: mealData
+    });
+  },
+  addAMeal(req, res) {
+    const { name, size, price, summary, imageUrl } = req.body;
+
+    const newlyCreatedMeal = {
+      id: mealData[mealData.length - 1].id + 1,
+      name,
+      size,
+      price,
+      summary,
+      imageUrl
+    };
+
+    const data = mealData.push(newlyCreatedMeal);
+    if (data) {
+      return res.status(201).json({
+        status: 201,
+        message: "New meal has been added",
+        data: [newlyCreatedMeal]
+      });
+    }
+
+    return res.status(500).json({
+      status: 500,
+      message: "something went wrong while trying to save your data"
+    });
+  },
+
+  getSingleMeal(req, res) {
+    const id = parseInt(req.params.id, 10);
+
+    mealData.map(mealData => {
```

Comment:
Since you are looking for a single element in the array, using the array method [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) would be significantly faster, and we'd have less code